sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/PrivateService.java

```diff
@@ -137,7 +91,13 @@ public static <T> Optional<T> extractCompasPrivate(TBaseElement baseElement, Cla
      * @throws ScdException throws when inconsistency between types
      */
     public static Optional<TCompasICDHeader> extractCompasICDHeader(TPrivate tPrivate) throws ScdException {
-        return extractCompasPrivate(tPrivate, TCompasICDHeader.class);
+        return getPrivateStream(List.of(tPrivate), TCompasICDHeader.class)
+                .reduce((tCompasICDHeader1, tCompasICDHeader2) -> toOneCompasICDHeader(COMPAS_ICDHEADER));
+    }
+
+    private static <T> T toOneCompasICDHeader(PrivateEnum privateEnum) {
```

Comment:
I don't think `toOneCompasICDHeader` is the correct name to choose, it takes PrivateEnum as an argument, so it will cover all kind of Private not only CompasICDHeader.
It was used in both `extractCompasICDHeader` and  `extractCompasPrivates`, I find that unclear.